pykickstart/commands/timesource.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use, modify,
+# copy, or redistribute it subject to the terms and conditions of the GNU
+# General Public License v.2.  This program is distributed in the hope that it
+# will be useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  Any Red Hat
+# trademarks that are incorporated in the source code or documentation are not
+# subject to the GNU General Public License and may only be used or replicated
+# with the express permission of Red Hat, Inc.
+#
+from pykickstart.errors import KickstartParseError
+from pykickstart.version import F33
+from pykickstart.base import BaseData, KickstartCommand
+from pykickstart.options import KSOptionParser
+
+from pykickstart.i18n import _
+
+class F33_TimesourceData(BaseData):
+    removedKeywords = BaseData.removedKeywords
+    removedAttrs = BaseData.removedAttrs
+
+    def __init__(self, *args, **kwargs):
+        BaseData.__init__(self, *args, **kwargs)
+        self.ntp_server = kwargs.get("ntp_server", "")
+        self.ntp_pool = kwargs.get("ntp_pool", "")
+        self.ntp_disable = kwargs.get("ntp_disable", False)
+        self.nts = kwargs.get("nts", False)
+
+    def __eq__(self, y):
+        if not y:
+            return False
+
+        if self.ntp_server:
+            return self.ntp_server == y.ntp_server and self.nts == y.nts
+        elif self.ntp_pool:
+            return self.ntp_pool == y.ntp_pool and self.nts == y.nts
+        else:
+            # given that one of ntp_server/ntp_pool/ntp_disable is required
+            # this will be --ntp_disable & such timesource invocations will
+            # always be the same by definition
+            return True
```

Comment:
I don't see how this could always be true? The assumption about `self` being `--ntp_disable` seems correct to me, but not the one about `y` being the same...